pkg/endpoint/endpoint.go

```diff
@@ -482,30 +489,45 @@ func (e *Endpoint) WaitForProxyCompletions(proxyWaitGroup *completion.WaitGroup)
 	}
 
 	start := time.Now()
-	e.getLogger().Debug("Waiting for proxy updates to complete...")
+
+	e.Mutex.RLock()
+	logger := e.getLogger()
+	e.Mutex.RUnlock()
+
```

Comment:
The added locking here is no longer necessary, as `e.getLogger()` takes care of it's own locking.